app/models/domain.rb

```diff
@@ -7,20 +7,13 @@ class Domain < ApplicationRecord
   validates :banned_reason, :length => { :maximum => 200 }
 
   validates :domain, presence: true
-  validates :is_tracker, inclusion: { in: [true, false] }
 
   def ban_by_user_for_reason!(banner, reason)
-    self.banned_at = Time.current
-    self.banned_by_user_id = banner.id
-    self.banned_reason = reason
-    self.save!
+    banning_switcher(:on, banner, reason)
```

Comment:
I really don't like this style of code sharing, where `banning_switcher` toggles between the two. For such sensitive functionality, I would rather have simpler, redundant code, like how `app/models/user.rb` implements `ban_by_user_for_reason!` and `unban_by_user!`. They are explicit and obvious about setting all the state they care about.

Reply:
OK. I will change it

Reply:
Done